### ISSUE TITLE
automatic glusterfs mount fails

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#1040]
Originally reported of Google Code with ID 984.

```
What software version are you running? Please provide the output of "gnt-
cluster --version", "gnt-cluster version", and "hspace --version".

gnt-cluster (ganeti v2.11.6) 2.11.6

Software version: 2.11.6
Internode protocol: 2110000
Configuration format: 2110000
OS api version: 20
Export interface: 0
VCS version: (ganeti) version v2.11.6

hspace (ganeti) version v2.11.6
compiled with ghc 7.6
running on linux x86_64


<b>What distribution are you using?</b>

Ubuntu 14.04 with packages from ppa pkg-ganeti-devel/lts

<b>What steps will reproduce the problem?</b>
1. Setup a ganeti cluster with a glusterfs VM storage
2. Test with an instance (gnt-instance add -t gluster …)
3. Get a crash when trying to create disk image

<b>What is the expected output? What do you see instead?</b>

Should continue with the VM guest installation.

Log shows:
2014-09-16 15:10:16,380: ganeti-noded pid=28631 INFO RunCmd mount -t glusterfs 127.0.0.1:24007:img-data /var/run/ganeti/gluster
2014-09-16 15:10:16,448: ganeti-noded pid=28631 ERROR '/var/run/ganeti/gluster': mount failure: try running 'gluster volume info img-data' on 127.0.0.1 to ensure it exists, it is started and it is using the tcp transport

<b>Please provide any additional information below.</b>

It's just the "mount" command that fails. No port can be set:
        mount -t glusterfs [-o <options>] <volumeserver>:<volumeid> <mountpoint>

It may be a change in glusterfs itself, not sure. I'm running version: 3.4.2-1ubuntu1

Currently I just removed the port setting in the code:
  file: /usr/share/ganeti/2.11/ganeti/storage/gluster.py
  class: GlusterVolume
  method: _GetFUSEMountString
remove all about "port" (two locations)

```
Originally added on 2014-10-31 15:18:53 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-10-31 15:59:25 +0000 UTC.

Added Labels: Component-storage

**Comment by Ganeti-Issues-Migrator** [comment]
```
My volume name exists in the gluster, the volume is started, iptables has the necessary ports open, ganeti appears to be properly configured, but when I go to create an instance with disk template type of "gluster", it fails to recognize the existence of the volume. Is there something that I'm missing? I've tried kernelspace and userspace access modes. i'm also on version 2.12.0 of ganeti and glusterfs 3.5.2.


gnt-cluster info:
...
  gluster: 
    access: userspace
    host: 10.66.8.201
    port: 24007
    volume: gv0
...


root@node3:/home/ivoadmin# gnt-instance add -t gluster --disk 0:size=1024 --no-ip-check --no-name-check -o debootstrap+default glustertest
Sun Dec  7 15:59:57 2014  - INFO: Selected nodes for instance glustertest via iallocator hail: node3.ivo.net
Sun Dec  7 16:00:01 2014 * disk 0, size 1.0G
Sun Dec  7 16:00:01 2014 * creating instance disks...
Sun Dec  7 16:00:01 2014  - WARNING: Device creation failed
Failure: command execution error:
Can't create block device <Disk(type=gluster, logical_id=('loop', 'ganeti/b7a4d55d-9099-4d90-9225-ca07675606d2.0'), children=None, visible as /dev/disk/0, size=1024m)> on node node3.ivo.net for instance glustertest: Can't create block device: '/var/run/ganeti/gluster': mount failure: try running 'gluster volume info gv0' on 10.66.8.201 to ensure it exists, it is started and it is using the tcp transport


root@gluster1:/home/ivoadmin# gluster volume info gv0
 
Volume Name: gv0
Type: Replicate
Volume ID: f0a4b51b-c48c-4c6a-bf31-e8b2206f3fb0
Status: Started
Number of Bricks: 1 x 2 = 2
Transport-type: tcp
Bricks:
Brick1: gluster1:/export/sdb1/brick
Brick2: gluster2:/export/sdb1/brick
Options Reconfigured:
auth.allow: 10.66.8.103
server.allow-insecure: on
root@gluster1:/home/ivoadmin# 


root@gluster1:/home/ivoadmin# iptables -L
Chain INPUT (policy ACCEPT)
target     prot opt source               destination         
ACCEPT     tcp  --  anywhere             anywhere             tcp dpt:http
ACCEPT     tcp  --  anywhere             anywhere             tcp dpts:24007:24047
ACCEPT     tcp  --  anywhere             anywhere             tcp dpt:sunrpc
ACCEPT     udp  --  anywhere             anywhere             udp dpt:sunrpc

I'll be happy to provide any other info necessary.

Thanks

```
Originally added on 2014-12-07 21:28:14 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-12-08 11:23:24 +0000 UTC.

Added Labels: Type-Defect
Changed State: Accepted
Added to Milestone: Release2.14

**Comment by Ganeti-Issues-Migrator** [comment]
```
Unfortunately I can't verify this soon, as I do not have a GlusterFS cluster handy, and I probably won't be able to set one up before Christmas holidays start up.

Dimitris, who relatively recently provided some gluster patches, might be able to help out, so I am cc-ing him.


```
Originally added on 2014-12-09 14:31:56 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Fixed in:

commit 8d9a0b58575da4ea609856adc92cc1a0a56446b9
Author: Dimitris Bliablias <bl.dimitris@gmail.com>
Date:   Mon Jan 5 17:09:33 2015 +0200

    Gluster: fix mount.glusterfs argument passing
    
    Currently, the Gluster server port is included in the volume path,
    i.e., <host>:<port>:<volumeid>, but the mount.glusterfs script requires
    to explicitly specify the server port using the '-o server-port=port'
    option. This would cause the port setting to be ignored and a default of
    24007 to be used when using glusterfs versions later than 3.3.0 with
    Ganeti, while for newer versions glusterfs mount fails.
    
    This patch modifies '_GetFUSEMountString' method to generate the correct
    argument string for the mount.glusterfs script. Also, it updates the
    'ganeti.storage.gluster_unittest.py' unit test to correspond to the new
    changes.
    
    This fixes issue #984.
    
    Signed-off-by: Dimitris Bliablias <bl.dimitris@gmail.com>
    Reviewed-by: Hrvoje Ribicic <riba@google.com>
```
Originally added on 2015-01-21 14:07:43 +0000 UTC.

Changed State: Fixed
Added to Milestone: Release2.11
Removed from Milestone: Release2.14

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released as part of 2.14.0
```
Originally added on 2015-06-02 10:39:05 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released as part of 2.14.0
```
Originally added on 2015-06-02 10:39:42 +0000 UTC.

Changed State: Released
